widgets/fanspeed_freebsd.lua

```diff
@@ -0,0 +1,27 @@
+--
+-- fanspeed: provides speed level of main fan
+--
+-- expects one (1) full sysctl string to entry
+--   e.g.: "dev.acpi_ibm.0.fan_speed"
+
+-- environment
+local setmetatable = setmetatable
+local string = { match = string.match }
+local helpers = require("vicious.helpers")
+
+local fanspeed_freebsd = {}
+
+local function worker(format, warg)
+    if not warg then return end
+
+    local fanspeed = helpers.sysctl( "" .. warg .. "" )
+
+    if not fanspeed then
+        -- use negative fanspeed to indicate error
+        return {-1}
+    else
+        return {string.match(fanspeed, "[%d]+")}
```

Comment:
There must have been reason to solve it that way. But i can't remeber it.
I will ponder over this some more, but your argument is very valid.